src/app/view/editor-main-view/data-views/trainrunSectionViewObject.ts

```diff
@@ -51,6 +52,10 @@ export class TrainrunSectionViewObject {
       .filter((section) => !section.getSourceNode().isNonStop(section)).length;
   }
 
+  getCollapsedNodeFromStopIndex(stopIndex: number): Node {
+    return this.trainrunSections[stopIndex].getTargetNode();
```

Comment:
Not all collapsed nodes are stops: I think we need to skip any non-stop node here.